spec/lib/telephony/otp_sender_spec.rb

```diff
@@ -0,0 +1,250 @@
+    let(:domain) { 'login.gov' }
+    let(:country_code) { 'US' }
+
+    before do
+      allow(Telephony.config).to receive(:adapter).and_return(:test)
+    end
+
+    context 'for SMS' do
+      let(:channel) { :sms }
+
+      it 'saves the OTP that was sent for authentication' do
+        subject.send_authentication_otp
+
+        expect(Telephony::Test::Message.last_otp).to eq(otp)
+      end
+
+      it 'saves the OTP that was sent for confirmation' do
+        subject.send_confirmation_otp
+
+        expect(Telephony::Test::Message.last_otp).to eq(otp)
+      end
+    end
+
+    context 'for Voice' do
+      let(:channel) { :voice }
+
+      it 'saves the OTP that was sent for authentication' do
+        subject.send_authentication_otp
+
+        expect(Telephony::Test::Call.last_otp).to eq(otp)
+      end
+
+      it 'saves the OTP that was sent for confirmation' do
+        subject.send_confirmation_otp
+
+        expect(Telephony::Test::Call.last_otp).to eq(otp)
+      end
+    end
+  end
+
+  context 'with the pinpoint adapter' do
+    subject do
+      described_class.new(
+        to: to,
+        otp: otp,
+        expiration: expiration,
+        channel: channel,
+        domain: domain,
+        country_code: country_code,
+      )
+    end
+
+    let(:to) { '+1 (202) 262-1234' }
+    let(:otp) { '123456' }
+    let(:expiration) { 5 }
+    let(:domain) { 'login.gov' }
+    let(:country_code) { 'US' }
+
+    before do
+      allow(Telephony.config).to receive(:adapter).and_return(:pinpoint)
+    end
+
+    context 'for SMS' do
+      let(:channel) { :sms }
+
+      it 'sends an authentication OTP with Pinpoint SMS' do
+        message = "Login.gov: Your security code is 123456. "\
+                  "It expires in 5 minutes. Don't share this "\
+                  "code with anyone.\n\n@login.gov #123456"
+
+        adapter = instance_double(Telephony::Pinpoint::SmsSender)
+        expect(adapter).to receive(:send).with(
+          message: message,
+          to: to,
+          otp: otp,
+          country_code: 'US',
+        )
+        expect(Telephony::Pinpoint::SmsSender).to receive(:new).and_return(adapter)
+
+        subject.send_authentication_otp
+      end
+
+      it 'sends a confirmation OTP with Pinpoint SMS' do
+        message = "Login.gov: Your security code is 123456. It "\
+        "expires in 5 minutes. Don't share this code with anyone."\
+        "\n\n@login.gov #123456"
+
+        adapter = instance_double(Telephony::Pinpoint::SmsSender)
+        expect(adapter).to receive(:send).with(
+          message: message,
+          to: to,
+          otp: otp,
+          country_code: 'US',
+        )
+        expect(Telephony::Pinpoint::SmsSender).to receive(:new).and_return(adapter)
+
+        subject.send_confirmation_otp
+      end
+    end
+
+    context 'for voice' do
+      let(:channel) { :voice }
+
+      it 'sends an authentication OTP with Pinpoint Voice' do
+        message = <<~XML.strip.gsub(/\s+/, ' ')
```

Comment:
another `squish` candidate

```suggestion
        message = <<~XML.squish
```